w06-ruby-on-rails/d3-dawn-ruby-oop/README.md

```diff
@@ -0,0 +1,190 @@
+
+**Challenge:**
+How would we create an instance method for Car named `color?`, which tests whether the color matches what is passed in?
+
+```ruby
+alpha = Car.new
+alpha.color = "orange"
+alpha.color?("blue")
+=> false
+alpha.color?("orange")
+=> true
+```
+
+*Hint: define a new method with the `def` keyword inside the class*
+
+##Initialization & Instance Variables (10m)
+
+**Challenge:**
+What If I want to set the color of the car at the moment I'm creating it? How could I enable code like this?
+
+```ruby
+lambo = Car.new("red")
+lambo.color
+=> "red"
+```
+
+*Hint: Create an instance variable with `@`*
+
+##Class Variables (10m)
+
+What if I wanted a running counter for all the cars I've ever created? Enable the code: `Car.total` which outputs the total number of all cars ever created.
+
+*Hint: Create a class variable with `@@` to store the count and use the `self` keyword to add a class method*
+
+*Note: Class variables are used much more infrequently than instance variables*
+
+## Quick Review (15m)
+
+Think, pair, share...
+
+  * What is a class?
+    - What is an attribute?
+    - What is a method?
+  * What is the difference between:
+    - an instance variable,
+    - a class variable
+  * Why do we use classes?
+  * What is inheritance?
+
+## Inheritance (25m)
+
+Given an `Vehicle` class that contains the method `accelerate` & attribute `speed`..
+
+```ruby
+class Vehicle
+  attr_accessor :speed
+  def initialize
+  	@speed = 0
+  end
+  def accelerate(speed)
+  	@speed += speed
+  end
+end
+```
+
+**Challenge:**
+How can I create both `Car` and `Motorbike` classes while being DRY and not duplicating the method `accelerate` in each?
+
+## Exercise: Characters and Superheroes
+
+Previously you worked with using inheritance to create some `Animal` and `People` classes. This exercise will review those concepts, using the `Character` and `Superhero` classes.
+
+Superheroes are still Characters after all. In this exercise, you'll define:
+
+  1. A `Character` class, with the following:
+    * Attributes (instance variables defined in initialize method):
+      * `name`: A string that holds the character's name
+    * Instance Methods:
+      * `location`: Takes a parameter `origin` prints out a message telling the character's `name` and where they are from (their `origin`)
+      * `exercise` & `rest`: These two methods should NOT be passed any arguments. Instead, they will set an instance variable `@state` to the string `"ready"` or `"tired"` respectively.
+    - *Note:* What type of `attr` method will be needed to `read` the name and state variables?
+
+  2. A `Superhero` class, with the following characteristics:
+    * Inherits from `Character`
+    * Adds 2 new instance variables (which initialize takes as arguments):
+      * `charisma`
+      * `secret_identity`
+    * initializes using `super`, passing `name` as an argument
+    - Adds 2 new methods:
+      - `persuade`, which
+        - takes another character as an argument
```

Comment:
Went with "another instance of the `Superhero` class", as current implementation of `Character`'s don't have `charisma`. Changed "taco" name for `persuade` method to `other_superhero` instead of `other_character`.